csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs

```diff
@@ -309,92 +316,139 @@ private void TestMultiThreads()
+                { "mask_rcnn_keras", "Length of memory (3145728) must match product of dimensions (3)" },
+                { "mlperf_resnet", "Value cannot be null" },
+                { "mlperf_ssd_mobilenet_300", "Could not find file output_0.pb" },
+                { "yolov3", "Length of memory(519168) must match product of dimensions(3)" }
+            };
 
             var disableContribOpsEnvVar = Environment.GetEnvironmentVariable("DisableContribOps");
             var isContribOpsDisabled = (disableContribOpsEnvVar != null) ? disableContribOpsEnvVar.Equals("ON") : false;
             if (isContribOpsDisabled)
             {
-                skipModels.Add("test_tiny_yolov2");
+                skipModels["test_tiny_yolov2"] =  "Fails when ContribOps is disabled";
             }
 
-            var opsets = new[] { "opset7", "opset8" };
+            return skipModels;
+        }
+
+        public static IEnumerable<object[]> GetModelsForTest()
+        {
+            var modelsDir = GetTestModelsDir();
+            var modelsDirInfo = new DirectoryInfo(modelsDir);
+            var skipModels = GetSkippedModels();
+
+            foreach (var opsetDir in modelsDirInfo.EnumerateDirectories())
+            {
+                //var modelRoot = new DirectoryInfo(Path.Combine(modelsDir, opsetDir.Name));
+                foreach (var modelDir in opsetDir.EnumerateDirectories())
+                {
+                    if (!skipModels.ContainsKey(modelDir.Name))
+                    {
+                        yield return new object[] { modelDir.Parent.Name, modelDir.Name, modelDir.FullName };
+                    }
+                } //model
+            } //opset
+        }
+
+        public static IEnumerable<object[]> GetSkippedModelForTest()
+        {
             var modelsDir = GetTestModelsDir();
-            foreach (var opset in opsets)
+            var modelsDirInfo = new DirectoryInfo(modelsDir);
+            var skipModels = GetSkippedModels();
+
+            foreach (var opsetDir in modelsDirInfo.EnumerateDirectories())
             {
-                var modelRoot = new DirectoryInfo(Path.Combine(modelsDir, opset));
+                var modelRoot = new DirectoryInfo(Path.Combine(modelsDir, opsetDir.Name));
                 foreach (var modelDir in modelRoot.EnumerateDirectories())
                 {
-                    String onnxModelFileName = null;
+                    if (skipModels.ContainsKey(modelDir.Name))
+                    {
+                        //Console.WriteLine("Model {0} is skipped due to the error: {1}", modelDir.FullName, skipModels[modelDir.Name]);
+                        yield return new object[] { modelDir.Parent.Name, modelDir.Name, modelDir.FullName };
+                    }
+                        
+                }
+            }
+        }
+
 
-                    if (skipModels.Contains(modelDir.Name))
-                        continue;
+        [x64Theory]
```

Comment:
Do we still require `[x64Theory]`? It was designed to skip testing of pre-trained models on a 32-bit process because the C# API originally could not interface with 32-bit ORT. This can probably be removed, and the general `[Theory]` attribute should work.